### PR TITLE
Rdcc 739

### PR DESCRIPTION
**Pull request by reem-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-887

### Change description ###

Setting up pipeline

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java [hunk 1]
```diff
@@ -72,14 +70,16 @@ public void setUp() {
...
+        RestAssured.useRelaxedHTTPSValidation();
+
 
-        IdamOpenIdClient idamOpenIdClient = new IdamOpenIdClient(configProperties);
-        log.info("idamOpenIdClient: " + idamOpenIdClient);
         /*SerenityRest.proxy("proxyout.reform.hmcts.net", 8080);
         RestAssured.proxy("proxyout.reform.hmcts.net", 8080);*/
 
+        IdamOpenIdClient idamOpenIdClient = new IdamOpenIdClient(configProperties);
+        log.info("idamOpenIdClient: " + idamOpenIdClient);
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java]: logger do we needed?

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java [hunk 2]
```diff
@@ -31,8 +32,26 @@ public JudicialApiClient(String judicialApiUrl,
...
 
     public String getWelcomePage() {
+
+        ResponseBodyExtractionOptions extractionOptions = withUnauthenticatedRequest()
+                .when()
+                .get("/")
+                .body();
+
+        log.info("GET response body::::::" +  withUnauthenticatedRequest()
+                .get("/")
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java]: same as above. remove logger.

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java [hunk 3]
```diff
@@ -41,9 +60,24 @@ public String getWelcomePage() {
...
+
     }
 
     public String getHealthPage() {
+        ResponseBodyExtractionOptions extractionOptions = withUnauthenticatedRequest()
+                .when()
+                .get("/health")
+                .body();
+
+        log.info("GET response body::::::" + withUnauthenticatedRequest()
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java]: same as above

File: src/functionalTest/resources/application-functional.yaml [hunk 4]
```diff
@@ -14,7 +14,7 @@ idam.auth.clientId: rd-judicial-api
 
 idam.auth.tokenAuth: ${IDAM_TOKEN_ATUH:cmQtcHJvZmVzc2lvbmFsLWFwaTpjYzVmMmE2LTk2OTAtMTFlOS1iYzQyLTUyNmFmNzc2NGY2NA==}
 idam.auth.redirectUrl: ${OAUTH2_REDIRECT_URI:https://rd-judicial-api-aat.service.core-compute-aat.internal/oauth2redirect}
-s2s-secret: ${S2S_SECRET:K6KA6PRLRFWQLSD6}
+s2s-secret: ${S2S_SECRET:F5U22E7HCOOEZY5P}
```
Comment (abhijitdiwan) [inline src/functionalTest/resources/application-functional.yaml]: Please replace with dummy value AAAAAAAAAAAAAAAA

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -66,16 +66,15 @@ security:
...
-    - rd-judicial-api
+    - rd_judicial_api
 
 
 ### dependency configuration
 auth.provider.service.client.baseUrl: ${S2S_URL:http://rpe-service-auth-provider-aat.service.core-compute-aat.internal}
 idam:
   s2s-auth:
-    totp_secret: ${S2S_SECRET:K6KA6PRLRFWQLSD6}
+    totp_secret: ${S2S_SECRET:F5U22E7HCOOEZY5P}
```
Comment (abhijitdiwan) [inline src/main/resources/application.yaml]: Please replace with dummy value.

File: config/owasp/suppressions.xml [hunk 6]
```diff
@@ -1517,37 +1551,1593 @@
...
+   file name: phantomjsdriver-1.4.4.jar
+   ]]></notes>
+        <sha1>e5778e3273f8201bf9e0ff65fb8118f37b84beba</sha1>
+        <cve>CVE-2019-17221</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
+   file name: jackson-databind-2.9.8.jar
+   ]]></notes>
+        <packageUrl regex="true">^pkg:maven/com\.fasterxml\.jackson\.core/jackson\-databind@.*$</packageUrl>
```
Comment (abhijitdiwan) [inline config/owasp/suppressions.xml]: do we need all these suppressions?

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java (hunk 1 above)
Comment (reem-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java]: I may keep this for now as IDAM integration is the next step in development work for this service, but I'm happy to remove logging thereafter

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java (hunk 2 above)
Comment (reem-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java]: Done

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java (hunk 3 above)
Comment (reem-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/client/JudicialApiClient.java]: Done.

File: src/functionalTest/resources/application-functional.yaml (hunk 4 above)
Comment (reem-hmcts) [reply src/functionalTest/resources/application-functional.yaml]: Done

File: src/main/resources/application.yaml (hunk 5 above)
Comment (reem-hmcts) [reply src/main/resources/application.yaml]: Done

File: config/owasp/suppressions.xml (hunk 6 above)
Comment (reem-hmcts) [reply config/owasp/suppressions.xml]: I have gone through them and removed the unnecessary suppressions.

**Pull request closed without merging**